xmr-btc/tests/e2e.rs

```diff
@@ -0,0 +1,386 @@
+    let (b_sender, a_receiver): (Sender<bob::Message>, Receiver<bob::Message>) = mpsc::channel(5);
+
+    let a_transport = Transport {
+        sender: a_sender,
+        receiver: a_receiver,
+    };
+
+    let b_transport = Transport {
+        sender: b_sender,
+        receiver: b_receiver,
+    };
+
+    (a_transport, b_transport)
+}
+
+pub async fn init_test<'a>(
+    monero: &'a Monero<'a>,
+    bitcoind: &Bitcoind<'_>,
+) -> (
+    alice::State0,
+    bob::State0,
+    AliceNode<'a>,
+    BobNode<'a>,
+    InitialBalances,
+    SwapAmounts,
+) {
+    // must be bigger than our hardcoded fee of 10_000
+    let btc_amount = bitcoin::Amount::from_sat(10_000_000);
+    let xmr_amount = monero::Amount::from_piconero(1_000_000_000_000);
+
+    let swap_amounts = SwapAmounts {
+        xmr: xmr_amount,
+        btc: btc_amount,
+    };
+
+    let fund_alice = TEN_XMR;
+    let fund_bob = 0;
+    monero.init(fund_alice, fund_bob).await.unwrap();
+
+    let alice_monero_wallet = wallet::monero::AliceWallet(&monero);
+    let bob_monero_wallet = wallet::monero::BobWallet(&monero);
+
+    let alice_btc_wallet = wallet::bitcoin::Wallet::new("alice", &bitcoind.node_url)
+        .await
+        .unwrap();
+    let bob_btc_wallet = wallet::bitcoin::make_wallet("bob", &bitcoind, btc_amount)
+        .await
+        .unwrap();
+
+    let (alice_transport, bob_transport) = init_alice_and_bob_transports();
+
+    let alice = AliceNode::new(alice_transport, alice_btc_wallet, alice_monero_wallet);
+
+    let bob = BobNode::new(bob_transport, bob_btc_wallet, bob_monero_wallet);
+
+    let alice_initial_btc_balance = alice.bitcoin_wallet.balance().await.unwrap();
+    let bob_initial_btc_balance = bob.bitcoin_wallet.balance().await.unwrap();
+
+    let alice_initial_xmr_balance = alice.monero_wallet.0.get_balance_alice().await.unwrap();
+    let bob_initial_xmr_balance = bob.monero_wallet.0.get_balance_bob().await.unwrap();
+
+    let redeem_address = alice.bitcoin_wallet.new_address().await.unwrap();
+    let punish_address = redeem_address.clone();
+    let refund_address = bob.bitcoin_wallet.new_address().await.unwrap();
+
+    let alice_state0 = alice::State0::new(
+        &mut OsRng,
+        btc_amount,
+        xmr_amount,
+        RELATIVE_REFUND_TIMELOCK,
+        RELATIVE_PUNISH_TIMELOCK,
+        redeem_address.clone(),
+        punish_address.clone(),
+    );
+    let bob_state0 = bob::State0::new(
+        &mut OsRng,
+        btc_amount,
+        xmr_amount,
+        RELATIVE_REFUND_TIMELOCK,
+        RELATIVE_PUNISH_TIMELOCK,
+        refund_address,
+    );
+    let initial_balances = InitialBalances {
+        alice_xmr: alice_initial_xmr_balance,
+        alice_btc: alice_initial_btc_balance,
+        bob_xmr: bob_initial_xmr_balance,
+        bob_btc: bob_initial_btc_balance,
+    };
+    (
+        alice_state0,
+        bob_state0,
+        alice,
+        bob,
+        initial_balances,
+        swap_amounts,
+    )
+}
+
+#[cfg(test)]
```

Comment:
That is redundant within the `tests/` directory. All of this code gets only compiled if the `test` cfg is set anyway.